nucypher-core/tests/test_encryption_vectors.rs

```diff
@@ -0,0 +1,305 @@
+#[cfg(test)]
+mod tests {
+    use chacha20poly1305::aead::Aead;
+    use chacha20poly1305::{ChaCha20Poly1305, Key, KeyInit, Nonce};
+    use rand::rngs::OsRng;
+    use rand::RngCore; // Add this import for the fill_bytes method
+    use serde::{Deserialize, Serialize};
+    use serde_json::{json, Value};
+    use std::fs;
+    use std::path::Path;
+
+    // Since the dkg module is private, we reimplement the encryption/decryption functions here
+    // based on the implementation in src/dkg.rs
+
+    // Structure that matches the JSON test vector format
+    #[derive(Serialize, Deserialize)]
+    struct TestVector {
+        id: String,
+        description: String,
+        shared_secret: Vec<u8>,
+        plaintext: Option<String>,
+        fixed_nonce: Option<Vec<u8>>,
```

Comment:
This could be annotated with `#[serde(with = "serde_bytes::as_hex")]` similarly to bytestring fields in the main code, this way the test vectors will look a little tidier

Reply:
Also, just FYI, the `umbral_pre::serde_bytes` (that I wrote back in the day) has since been extracted into https://docs.rs/serde-encoded-bytes